src/repo-service.cpp

```diff
@@ -138,3 +295,132 @@ void RepoService::openLocalFile(const QString& repo_id,
         helper->start();
     }
 }
+
+void RepoService::onGetRequestSuccess(const ServerRepo& repo)
+{
+    GetRepoRequest* req = qobject_cast<GetRepoRequest*>(sender());
+    if (!req)
+        return;
+
+    SyncedSubfolderhasRepoID subfolder_helper(repo.id);
+    std::vector<SyncedSubfolder>::iterator pos = std::find_if(synced_subfolders_.begin(), synced_subfolders_.end(), subfolder_helper);
+
+    if (pos != synced_subfolders_.end()) {
+        ServerRepo fixed_repo = repo;
+        fixed_repo.parent_path = pos->parent_path();
+        fixed_repo.parent_repo_id = pos->parent_repo_id();
+        server_repos_.push_back(fixed_repo);
+    }
+
+    delete req;
+    get_repo_reqs_.pop_front();
+
+    if (!get_repo_reqs_.empty())
+        get_repo_reqs_.front()->send();
+    else
+        emit refreshSuccess(server_repos_);
+}
+
+void RepoService::onGetRequestFailed(const ApiError& /*error*/)
+{
+    GetRepoRequest* req = qobject_cast<GetRepoRequest*>(sender());
+    if (!req)
+        return;
+
+    delete req;
+    get_repo_reqs_.pop_front();
+
+    if (!get_repo_reqs_.empty())
+        get_repo_reqs_.front()->send();
+    else
+        emit refreshSuccess(server_repos_);
+}
+
+void RepoService::startGetRequestFor(const QString &repo_id)
+{
+    bool was_empty = get_repo_reqs_.empty();
+    GetRepoRequest *req = new GetRepoRequest(seafApplet->accountManager()->currentAccount(), repo_id);
+    get_repo_reqs_.push_back(req);
+    connect(req, SIGNAL(success(const ServerRepo&)), this, SLOT(onGetRequestSuccess(const ServerRepo&)));
+    connect(req, SIGNAL(failed(const ApiError&)), this, SLOT(onGetRequestFailed(const ApiError&)));
+    if (was_empty)
```

Comment:
I think readable code is more important than the trivial performance gain. 

When I read this function, I see the `was_empty` variable is defined at the very beginning of the function, but used at the very end of the function, which is quite counter-intuitive, and makes me think:  "Why did he do this?" and takes me while to understand it.

Of course I'm not saying performance is not important, but considering today's cpu frequency, I really don't think we need to save cpu cycles everywhere for a desktop application at the cost of code readability.